AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/CreateKeysHV2.dfy

```diff
@@ -1,48 +1,61 @@
 // Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+include "../Model/AwsCryptographyKeyStoreAdminTypes.dfy"
 include "../../AwsCryptographyKeyStore/Model/AwsCryptographyKeyStoreTypes.dfy"
 include "../../../../AwsCryptographyPrimitives/Model/AwsCryptographyPrimitivesTypes.dfy"
-include "../../AwsCryptographyKeyStore/src/Structure.dfy"
-include "../../AwsCryptographyKeyStore/src/DefaultKeyStorageInterface.dfy"
-include "../../AwsCryptographyKeyStore/src/KMSKeystoreOperations.dfy"
-include "../../AwsCryptographyKeyStore/src/ErrorMessages.dfy"
-include "../../AwsCryptographicMaterialProviders/src/AwsArnParsing.dfy"
-include "../../AwsCryptographyKeyStore/src/KmsArn.dfy"
-include "SystemKey/ContentHandler.dfy"
 
-// TODO-HV-2-M1: WIP
 module {:options "/functionSyntax:4" } CreateKeysHV2 {
-  // TODO-HV-2-M1-FF: Group imports according to libraries
+  // Standard Library Imports
   import opened StandardLibrary
   import opened Wrappers
+  import opened Seq
+  import opened UInt = StandardLibrary.UInt
+  import Time
+    // SDK Imports
+  import DDB = Com.Amazonaws.Dynamodb
+  import KMS = Com.Amazonaws.Kms
+    // Crypto Primitives Imports
+  import AtomicPrimitives
+    // MPL Imports
+  import AwsArnParsing
+  import CanonicalEncryptionContext
+    // (Branch) Key Store Imports
   import Structure
   import DefaultKeyStorageInterface
   import KMSKeystoreOperations
   import ErrorMessages = KeyStoreErrorMessages
-
-  import opened Seq
-  import opened UInt = StandardLibrary.UInt
   import KeyStoreTypes = AwsCryptographyKeyStoreTypes
-  import Types = AwsCryptographyKeyStoreAdminTypes
-  import DDB = ComAmazonawsDynamodbTypes
-  import KMS = ComAmazonawsKmsTypes
-  import AwsArnParsing
   import KmsArn
-  import AtomicPrimitives
-  import CanonicalEncryptionContext
-  import Time
+    // (Branch) Key Store Admin Imports
   import HvUtils = HierarchicalVersionUtils
-  import CryptoTypes = AwsCryptographyPrimitivesTypes
-  import ContentHandler = SystemKey.ContentHandler
+  import Types = AwsCryptographyKeyStoreAdminTypes
+
+  datatype CryptoAndKMS = CryptoAndKMS(
+    kmsConfig: KeyStoreTypes.KMSConfiguration,
+    grantTokens: KMS.Types.GrantTokenList,
+    kms: KMS.Types.IKMSClient,
+    Crypto: AtomicPrimitives.AtomicPrimitivesClient
+  ) {
```

Comment:
why do we need this new datatype? could we add a comment?

Reply:
Because it lets me consolidate a number of related relationships and make formally verified statements about them.
This lets me:
1. avoid repeat code
2. ensure that the histories of the two clients do not collide